gdal/port/cpl_vsil_hdfs.cpp

```diff
@@ -209,6 +207,9 @@ VSIHdfsFilesystemHandler::Open( const char *pszFilename,
                                 const char *pszAccess,
                                 bool)
 {
+  if (poFilesystem == nullptr)
```

Comment:
should be protected by a mutex. And probably shall be done in Stat() as well

Reply:
Done.